pkg/testutil/sets.go

```diff
@@ -0,0 +1,54 @@
+package testutil
+
+type Set[K comparable] struct {
+	entities map[K]any
+}
+
+func NewSetFromMap[K comparable, V any](m map[K]V) *Set[K] {
+	entities := make(map[K]any, 0)
+	for k := range m {
+		entities[k] = true
+	}
+	return &Set[K]{
+		entities: entities,
+	}
+}
+
+func NewSetFromSlice[K comparable](s []K) *Set[K] {
+	entities := make(map[K]any, 0)
+	for _, k := range s {
+		entities[k] = true
+	}
+	return &Set[K]{
+		entities: entities,
+	}
+}
+
+func (s *Set[K]) Difference(other *Set[K]) (additions []K, deletions []K) {
```

Comment:
These test functions seem complicated enough to warrant their own test file. haha

Reply:
ha yes it does 😆